File: Backend/CarWebShop/CarWebShop/Models/Messages.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace CarWebShop.Models
{
    public class Messages
    {
        public int SenderID { get; set; }
        public int ReceiverID { get; set; }
        public bool IsNew { get; set; }
        public int InitialSenderID { get; set; }
        [NotMapped]
        public string ReceiverUsername { get; set; }
        [NotMapped]
        public string SenderUsername { get; set; }
        public int AdverID { get; set; }


        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int MessageID { get; set; }



        public string Message { get; set; }
        public DateTime DateSent { get; set; }
        public User Receiver { get; set; }
        public User Sender { get; set; }

        public Advertisement Advertisement { get; set; }

    }
}

File: Frontend/UsedCarsShopFrontend/src/app/Data Transfer Objects/Messages.ts
export class Messages{
    Messages = [
        {
            chatID:0,
            senderUsername:'',
            receiverUsername:'',
            adverID:0,
            message:'',
            dateSent:new Date(),
            isSelected:false,
            messageID:0,
            isNew: Boolean,
            initialSenderID:0,
            advertisement:{
                adverID: 0,
                userID: 0,
                adverName: '',
                carID: 0,
                price: 0,
                userDto: {
                  userID: 0,
                  firstName: '',
                  lastName: '',
                  userName: '',
                  phoneNumber: '',
                },
                carDto: {
                  carID: 0,
                  model: '',
                  brand: '',
                  year: '',
                  type: '',
                  fuelType: '',
                  propulsion:'',
                  engineVolume:'',
                  horsePower:'',
                  miliage:''
                },
                imagePaths:[{
                  imagePath:'',
                  AdverID:0
                }],
                favoritedByUserDto:[{
                  UserID:0,
                  AdverID:0
                }]
            }
        }
    ]
}